core/src/test/java/kafka/testkit/KafkaClusterTestKit.java

```diff
@@ -141,30 +146,23 @@ public KafkaClusterTestKit build() throws Exception {
                         "CONTROLLER://localhost:0");
                     props.put(KafkaConfig$.MODULE$.ControllerListenerNamesProp(),
                         "CONTROLLER");
-                    setupNodeDirectories(baseDirectory, node.metadataDirectory(),
-                        Collections.emptyList());
                     // Note: we can't accurately set controller.quorum.voters yet, since we don't
                     // yet know what ports each controller will pick.  Set it to an
                     // empty string for now as a placeholder.
                     props.put(KafkaConfig$.MODULE$.ControllerQuorumVotersProp(), "");
-                    KafkaConfig config = new KafkaConfig(props, false,
-                        OptionConverters.toScala(Optional.empty()));
+                    setupNodeDirectories(baseDirectory, node.metadataDirectory(), Collections.emptyList());
+                    KafkaConfig config = new KafkaConfig(props, false, OptionConverters.toScala(Optional.empty()));
 
                     String threadNamePrefix = String.format("controller%d_", node.id());
-                    LogContext logContext = new LogContext("[Controller id=" + node.id() + "] ");
-
-                    metaLogManager = new LocalLogManager(
-                        logContext,
-                        node.id(),
-                        sharedLogData,
-                        threadNamePrefix
-                    );
-                    metaLogManager.initialize();
-                    Kip500Controller controller = new Kip500Controller(
-                        nodes.controllerProperties(node.id()),
-                        config,
-                        metaLogManager,
-                        new MockRaftManager(),
+                    MetaProperties metaProperties = MetaProperties.apply(nodes.clusterId(),
+                            OptionConverters.toScala(Optional.empty()),
+                            OptionConverters.toScala(Optional.of(node.id())));
+                    TopicPartition metadataPartition = new TopicPartition(KafkaServer.metadataTopicName(), 0);
+                    KafkaRaftManager raftManager = new KafkaRaftManager(metaProperties, metadataPartition, config,
+                            Time.SYSTEM, new Metrics(), connectFutureManager.future);
```

Comment:
Do we want to share a Metrics instance between Kip500Controller and KafkaRaftManager?

Reply:
Good catch. Yea, it makes sense to use the same metrics instance here.